bin/checkstyle.sh

```
@@ -0,0 +1,186 @@
+# diff is default mode
+MODE=${MODE:-diff}
+#MODE=${MODE@L}
+
+case "$MODE" in
+"diff")	[[ ! -d "$TOOLDIR/.git" ]] && die "Not in a git repository" ;;
+"full")	;;
+*)	die "Unknown checking mode '$MODE'" ;;
+esac
+
+TEMPDIR=$(mktemp -d) || die "Could not create temporary directory"
+trap "rm -rf '$TEMPDIR'" exit
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)